### ISSUE TITLE
UAP target doesn't have package references properly generated/used

### ISSUE THREAD

**Issue by clairernovotny** [dotnet/sdk#477]
When trying to cross compile with a project that includes a `uap10.0` target, the csc task doesn't have the package references resolved correctly and put on the CSC task.

To repro, look at the `vs2017` branch of https://github.com/onovotny/Zeroconf/tree/vs2017

In there you'll see it cross-compiling to `netstandard1.3` and `uap10.0`. The problem is that it's using the wrong `mscorlib` reference for `uap10.0` and when you look at the msbuild log and csc task invocation, it's not passing the correct /reference's that include all of the package dependencies that come from both the `NETStandard.Library` and the `Microsoft.NETCore.UniversalWindowsPlatform` reference. The RAR task seems to not generate anything.

Not sure if this is in the SDK tasks or the NuGet tasks?

**Comment by onovotny** [comment]
Happens on latest RC with preview4 sdk.

**Comment by onovotny** [comment]
Here is the diagnostic build log showing a netstandard and a uap cross compiled build. You can see that there are no resolved package dependencies for the uap build.

[build.txt](https://github.com/dotnet/sdk/files/647594/build.txt)

**Comment by davkean** [comment]
tag @nguerrera @ericstj

**Comment by davkean** [comment]
and @dsplaisted

**Comment by dsplaisted** [comment]
I think you can work around this by setting the following when the `TargetFramework` property is `uap10.0`:

```xml
<NugetTargetMoniker>UAP,Version=v10.0</NugetTargetMoniker>
```

The problem is that without this, the `NuGetTargetMoniker` is being set to "UAP,Version=v10.0.10240", which isn't matching the target in the lock file.  [Microsoft.PackageDependencyResolution.targets](https://github.com/dotnet/sdk/blob/d8e9a98dafd629010c13c622a00f72a4343a839e/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.PackageDependencyResolution.targets#L46-L47) is explicitly using the first three segments of the `TargetPlatformMinVersion` when the `TargetFrameworkMoniker` is ".NETCore,Version=v5.0".  So there may be a reason that the target moniker is supposed to include the Windows SDK version, and we may need a different solution than to simply use the simpler version of NugetTargetMoniker as in the workaround.

**Comment by natidea** [comment]
Those lines were inherited from [Microsoft.NuGet.targets](https://github.com/NuGet/NuGet.BuildTasks/blob/dev/src/Microsoft.NuGet.Build.Tasks/Microsoft.NuGet.targets#L49-L50) and presumably their application is the same in this context. The full condition suggests that explicitly setting `UseTargetPlatformAsNuGetTargetMoniker` to false will result in `NuGetTargetMoniker` == `TargetFrameworkMoniker`.

**Comment by onovotny** [comment]
While setting `<NugetTargetMoniker>UAP,Version=v10.0</NugetTargetMoniker>` let the references go through correctly, it created another issue -- all of the references are now copied to the output directory.

This should only happen for an AppX, Xap or Exe. The UWP library project shouldn't have all of the NuGet references copied to its output directory (slow/size, etc).

**Comment by onovotny** [comment]
Similar issues with other TFM's. It's not calculating the version correctly in the lock file. In the lock file it was Version=v0.0

For `portable-win81+wpa81`:
```
  <PropertyGroup Condition="'$(TargetFramework)' == 'portable-win81+wpa81'">
    <TargetFrameworkIdentifier>.NETPortable</TargetFrameworkIdentifier>
    <TargetFrameworkVersion>v4.6</TargetFrameworkVersion>
    <TargetFrameworkProfile>Profile32</TargetFrameworkProfile>
    <DefaultLanguage>en-US</DefaultLanguage>
    <NugetTargetMoniker>.NETPortable,Version=v0.0,Profile=Profile32</NugetTargetMoniker>
    <LanguageTargets>$(MSBuildExtensionsPath32)\Microsoft\Portable\$(TargetFrameworkVersion)\Microsoft.Portable.CSharp.targets</LanguageTargets>
  </PropertyGroup>```

**Comment by nguerrera** [comment]
> ...  it created another issue -- all of the references are now copied to the output directory.

@onovotny I believe this is because the SDK currently only defaults $(CopyLocalLockFileAssemblies) to false for .NET Core or .NET Standard. Can you try setting this to false yourself? Or did swapping the LanguageTargets already resolve that?

**Comment by onovotny** [comment]
@nguerrera adding that line to disable the local lock file copy did the trick.

**Comment by onovotny** [comment]
When working on this bug, keep in mind that calculating the NuGet Moniker based on `TargetPlatformMinVersion` may not work if UAP is bumped up to `uap10.1` as it exists here: https://github.com/dotnet/corefx/commit/7f2271034c5cc5571805df1228f548077f089b81. The version of Windows itself, and thus the PlatformMinVersion is unlikely to change (or will change for marketing reasons independent of what the TFM means).

**Comment by dsplaisted** [comment]
@onovotny This would be fixed by your PR #889, right?

**Comment by onovotny** [comment]
Yes
